.github/workflows/publish-release.yml

```diff
@@ -0,0 +1,67 @@
+name: Publish Release
+
+on:
+  workflow_call:
+    secrets:
+      NPM_TOKEN:
+        required: true
+
+jobs:
+  publish-release:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.sha }}
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@4f647fc679bcd3b11499ccb42104547c83dabe96
+        with:
+          toolchain: stable
+      - name: Install wasm-pack
+        run: curl https://raw.githubusercontent.com/rustwasm/wasm-pack/a3a48401795cd4b3afe1d74568c93675a04f3970/installer/init.sh -sSf | sh -s -- -f
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@82a92a6e8fbeee089604da2575dc567ae9ddeaab
```

Comment:
I see that these steps are repeated, can they be moved into a reusable job?

Reply:
as far as I can tell, such steps are almost always repeated (the famous actions/checkout is in almost every workflow step). I guess it can be moved to as reusable job but even the template doesn't do that: https://github.com/MetaMask/metamask-module-template/blob/main/.github/workflows/publish-release.yml